### ISSUE TITLE
Tutorial

### ISSUE THREAD

**Issue by fahadshery** [bnosac/crfsuite#6]
Looks like I hit a jackpot for creating my own models for NER, chunking etc.
A small tutorial would be ideal :)

**Comment by jwijffels** [comment]
Please look at the package vignette.

**Comment by fahadshery** [comment]
I did look at it. I am still not sure how to open the ShinyApp that comes with the package. (Sorry for being thick) and How to create a training dataset. I ran your examples successfully though

**Comment by jwijffels** [comment]
Open the file in RStudio and press on the run document button in RStudio.

**Comment by jwijffels** [comment]
I can't reproduce this error. The app runs fine with the following shiny/flexdashboard version

```

> installed.packages()[c("shiny", "flexdashboard"), "Version"]         shiny flexdashboard        "1.0.5"     "0.5.1.1"
--
```

**Comment by fahadshery** [comment]
My versions are:

`> installed.packages()[c("shiny", "flexdashboard"), "Version"]
        shiny flexdashboard 
      "1.1.0"     "0.5.1.1" ` is there a way to install the similar version as yours?

**Comment by fahadshery** [comment]
installed same version as yours:`devtools::install_version("shiny", version = "1.0.5", repos = "http://cran.us.r-project.org")` still getting the erro

**Comment by jwijffels** [comment]
sure, get the package from cran and install it

**Comment by jwijffels** [comment]
If you can't manager to install the app, you need to have/create training data in the format shown in the help of `?merge.chunkrange`

**Comment by jwijffels** [comment]
works fine and tested on Windows/Mac/Linux.
Run as `rmarkdown::run(system.file(package = "crfsuite", "app", "annotation.Rmd"))`

**Comment by fahadshery** [comment]
genius! This is why it wasn't working! Now It is up and running! it is time to build custom models... I intend to use your StarSpacy package as well. What's the key difference between the two?

**Comment by fahadshery** [comment]
I would also suggest to add 2 things in your wiki/intro/ReadMe file:

1. Add this line in the intro (`rmarkdown::run(system.file(package = "crfsuite", "app", "annotation.Rmd"))`) if people want to use the app :) 
2. Add confusion matrix instead of `table(scores$label)` this will show how accurate the model is after training and how good it is performing on the test set. you just have to tweak `table(scores$label)` to `table(train$label,scores$label)`
3. Lastly, for people who can't run the app, just put an example training file. This will help them format their data in the format crfsuite expects. If you can't add an example training_set then just provide a similar format mentioned by openNLP i.e. <START:person> Mike <END> who is just 10 months old. 
4. He likes to play in the lounge .
5. Again, thank you for the package. Its truely looking amazing!

**Comment by sirhcdlanodcm** [comment]
Just a heads up - there's a small typo in the code to run the shiny app in the vignette:

> And run the app with
>` rmarkdown::run(file = system.file(package = "crfsuite", "app", "annnotation.Rmd"))`

Note the 3 n's in "annotation.Rmd". Barely worth mentioning - but for people like me who like to copy and paste without reading, it might be a nice fix ;)

Super excited to explore this package!

**Comment by jwijffels** [comment]
the crfsuite R package requires you to have a tokenised data.frame (1 row per token, in the right sequence of token occurrence) containing:

- the category you want to model in IOB format
- in the columns, categorical information for each token (e.g. pos tag / is the token capitalised / position of the token in the document / lemma / first 4 letters / last 4 letters / whichever you think is relevant in explaining the category you want to model) as well as information of the surrounding (as in before/after) tokens or surrounding ngrams 
- an identifier of a sequence (e.g. a document identifier)

Examples (aprroximately 7) of these data structures are provided in 

- the package vignette: https://cran.r-project.org/web/packages/crfsuite/vignettes/crfsuite-nlp.html
- the documentation of the functions `?crf, ?predict.crf, ?ner_download_modeldata, ?crf_cbind_attributes, ?merge.chunkrange`

May I ask you to first look there before raising issues.
